test/test.js

```diff
@@ -140,33 +109,9 @@ function collect(browserNames) {
         throw new Error(`Browser is not downloaded. Run 'npm install' and try to re-run tests`);
     }
 
-    const browserEnvironment = new Environment(browserName);
-    browserEnvironment.beforeAll(async state => {
-      state._logger = utils.createTestLogger(config.dumpLogOnFailure);
-      state.browser = await state.browserType.launch({...launchOptions, logger: state._logger});
-    });
-    browserEnvironment.afterAll(async state => {
-      await state.browser.close();
-      delete state.browser;
-      delete state._logger;
-    });
-    browserEnvironment.beforeEach(async(state, testRun) => {
-      state._logger.setTestRun(testRun);
-    });
-    browserEnvironment.afterEach(async (state, testRun) => {
-      state._logger.setTestRun(null);
-    });
-
-    const pageEnvironment = new Environment('Page');
-    pageEnvironment.beforeEach(async state => {
-      state.context = await state.browser.newContext();
-      state.page = await state.context.newPage();
-    });
-    pageEnvironment.afterEach(async state => {
-      await state.context.close();
-      state.context = null;
-      state.page = null;
-    });
+    const logger = utils.createTestLogger(config.dumpLogOnFailure);
+    const browserEnvironment = new BrowserEnvironment(browserType, launchOptions, logger);
```

Comment:
Should BrowserEnvironment create the logger itself?

Reply:
Done.